promql/testdata/functions.test

```diff
@@ -149,6 +149,38 @@ eval_fail instant at 0m label_replace(testmetric, "src", "", "", "")
 
 clear
 
+# Tests for label_join.
+load 5m
+  testmetric{src="a",src1="b",src2="c",dst="original-destination-value"} 0
+  testmetric{src="d",src1="e",src2="f",dst="original-destination-value"} 1
+
+# label_join joins all src values in order.
+eval instant at 0m label_join(testmetric, "dst", "-", "src", "src1", "src2")
+  testmetric{src="a",src1="b",src2="c",dst="a-b-c"} 0
+  testmetric{src="d",src1="e",src2="f",dst="d-e-f"} 1
+
+# label_join treats non existent src labels as empty strings.
+eval instant at 0m label_join(testmetric, "dst", "-", "src", "src3", "src1")
+  testmetric{src="a",src1="b",src2="c",dst="a--b"} 0
+  testmetric{src="d",src1="e",src2="f",dst="d--e"} 1
+
+# label_join does not overwrite the destination label if the resulting dst label is empty string
+eval instant at 0m label_join(testmetric, "dst", "", "emptysrc", "emptysrc1", "emptysrc2")
+  testmetric{src="a",src1="b",src2="c",dst="original-destination-value"} 0
```

Comment:
All tests looking good.
I can see the idea behind keeping the original value. I think it might be harder to reason about though and can cause unexpected results in some cases. Unsetting a label by setting it to the empty string is valid operation in general, as we decided that a label not existing and having an empty value is equivalent.